src/app/resources/iis_scripts/admin_api_util.ps1

```diff
@@ -66,27 +88,76 @@ function EnsureMember($group, $userOrGroup) {
+                throw "IIS Administration API is not found, port pinged: $adminAPIPort"
+            }
+            throw "Unexpected service status for IIS Administration API: ""$($service.Status)"""
+        }
+        throw $_
+    }
 }
 
 $service = Get-WmiObject win32_service | Where-Object {$_.name -eq $serviceName}
-if ($service.StartInfo.EnvironmentVariables -and $service.StartInfo.EnvironmentVariables["USE_CURRENT_DIRECTORY_AS_ROOT"] -and $service.StartInfo.WorkingDirectory) {
-    $workingDirectory = $service.StartInfo.WorkingDirectory
+if ($service) {
+    if ($service.StartInfo.EnvironmentVariables -and $service.StartInfo.EnvironmentVariables["USE_CURRENT_DIRECTORY_AS_ROOT"] -and $service.StartInfo.WorkingDirectory) {
+        $workingDirectory = $service.StartInfo.WorkingDirectory
+    } else {
+        $proc = Get-Process -id $service.ProcessId
+        $workingDirectory = GetIISAdminHome $proc
+    }
 } else {
-    $proc = Get-Process -id $service.ProcessId
-    $workingDirectory = Split-Path ($proc.Modules | Where-Object { $_.ModuleName -eq "Microsoft.IIS.Administration.dll" }).FileName
+    ## dev-mode support, no restart can be perfomed
+    Write-Verbose "Dev mode, scanning processes for IIS Admin API"
+    $devMode = $true
+    $workingDirectory = GetIISAdminHome (Get-Process -ProcessName dotnet)
 }
 
 $configLocation = [System.IO.Path]::Combine($workingDirectory, "config", "appsettings.json")
+if ($devMode -and !(Test-Path $configLocation)) {
+    $configLocation = [System.IO.Path]::Combine($workingDirectory, "..", "..", "..", "config", "appsettings.json")
+}
+Write-Verbose "Config Location $configLocation"
 $config = Get-Content -Raw -Path $configLocation | ConvertFrom-Json
-
 $user = $(whoami)
-if (!$config.security.users.owners.Contains($user)) {
 
+$groupModified = $false
+
+## Install process adds $user in the config file. We don't want that, we want $iisAdminOwners in there instead
+if ($install) {
+    $config.security.users.owners = @()
+    $config.security.users.administrators = @()
+}
+
+if (!$config.security.users.owners.Contains($user)) {
     EnsureGroup $iisAdminOwners
-    EnsureMember $iisAdminOwners $user
+    $groupModified = EnsureMember $iisAdminOwners $user
 
     if (!$config.security.users.owners.Contains($iisAdminOwners)) {
+        $config.security.users.owners += $iisAdminOwners
+        $saveConfig = $true
+    }
+    if (!$config.security.users.administrators.Contains($iisAdminOwners)) {
+        $config.security.users.administrators += $iisAdminOwners
+        $saveConfig = $true
+    }
+
+    if ($saveConfig) {
+        Write-Verbose "Saving config..."
+        if ($devMode) {
+            throw "Cannot edit config file in dev mode, please add ""IIS Administration API Owners"" manually"
+        }
         if (!(Get-LocalGroupMember -Group "Administrators" -Member $user -ErrorAction SilentlyContinue)) {
             throw "Administrator privilege is needed to initiate IIS Administration API"

```

Comment:
Please add $user to the error message so that users know what user don't have administrative previledge.

Reply:
Updated